Compilation failure on the tester when compiling with clang (cgal 5.2.2)

I encounter the following compilation error on the tester for the last two weeks: [1, 2]
```
source/cgal/intersections.cc:719: error: no matching function for call to 'get_vertices_in_cgal_order'

      CGALWrappers::get_vertices_in_cgal_order<int(std::pow(2, dim0))>(
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<s>This doesn't seem to happen with CGAL 5.5.</s>

@luca-heltai @jh66637 A wild guess that this is related to 41630d07d67adc1361e9d18d7b3368add10fda16


[1] https://cdash.dealii.org/viewBuildError.php?buildid=227
[2] https://github.com/dealii/dealii/blob/master/source/cgal/intersections.cc#L719

Minimal reproducer:
```
#include <cmath>

int main()
{
  constexpr int dim = 1;
  constexpr int size = std::pow(2, dim);
}
```
gcc compiles this expression, clang correctly complains:
```
test.cc:6:17: error: constexpr variable 'size' must be initialized by a constant expression
  constexpr int size = std::pow(2, dim);
                ^      ~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/include/g++-v12/cmath:418:14: note: non-constexpr function 'pow' cannot be used in a constant expression
      return pow(__type(__x), __type(__y));
             ^
test.cc:6:24: note: in call to 'pow(2, 1)'
  constexpr int size = std::pow(2, dim);
                       ^
/usr/include/bits/mathcalls.h:140:17: note: declared here
__MATHCALL_VEC (pow,, (_Mdouble_ __x, _Mdouble_ __y));
                ^
1 error generated.
```

--- a/source/cgal/intersections.cc
+++ b/source/cgal/intersections.cc
@@ -22,6 +22,7 @@
 #ifdef DEAL_II_WITH_CGAL
 
 #  include <deal.II/base/quadrature_lib.h>
+#  include <deal.II/base/utilities.h>
 
 #  include <deal.II/fe/mapping.h>
 
@@ -716,19 +717,19 @@ namespace CGALWrappers
            ExcNotImplemented());
 
     const auto vertices0 =
-      CGALWrappers::get_vertices_in_cgal_order<int(std::pow(2, dim0))>(
+      CGALWrappers::get_vertices_in_cgal_order<Utilities::pow(2, dim0)>(
         cell0, mapping0);
     const auto vertices1 =
-      CGALWrappers::get_vertices_in_cgal_order<int(std::pow(2, dim1))>(
+      CGALWrappers::get_vertices_in_cgal_order<Utilities::pow(2, dim1)>(
         cell1, mapping1);
 
     return compute_intersection_of_cells<dim0,
                                          dim1,
                                          spacedim,
-                                         int(std::pow(2, dim0)),
-                                         int(std::pow(2, dim1))>(vertices0,
-                                                                 vertices1,
-                                                                 tol);
+                                         Utilities::pow(2, dim0),
+                                         Utilities::pow(2, dim1)>(vertices0,
+                                                                  vertices1,
+                                                                  tol);
   }
 
 #  include "intersections.inst"

